game/src/main/java/net/demilich/metastone/game/spells/PutDeckTopSpell.java

```diff
@@ -21,7 +21,11 @@ protected void onCast(GameContext context, Player player, SpellDesc desc, Entity
 		CardList cards = SpellUtils.getCards(context, player, target, source, desc);
 
 		for (Card card : cards) {
-			context.getLogic().putOnTopOfDeck(player, card);
+			try {
+				context.getLogic().putOnTopOfDeck(player, card);
```

Comment:
What issue was this addressing? This is a real bug if its already in the location. Then a copy wasn't made. Probably the duplicate card on top of your deck spell right?

Better to just make a custom spell for that.